app/ovpn/get-ovpn-files.sh

```diff
@@ -2,26 +2,42 @@
 
 . /app/date.sh --source-only
 
-echo "$(adddate) INFO: Download and extract ovpn files"
+# check if the file exists
+if [ -f ${OVPN_CONFIG_DIR}/ovpn.zip ]; then
+  #the file exists continue checking if its older than two hours.
+  if test `find ${OVPN_CONFIG_DIR}/ovpn.zip -mmin +120` then
+    download_files()
+  else
+    echo "$(adddate) INFO: Skipping downloading OVPN files - as they are not older than two hours."
+  fi
+else
+  #the files don't exists contiue to download and extract them.
+  download_files()
+fi
 
-#Create directory if no volume is done
-mkdir -p /app/ovpn/config
+function download_files() {
+  # the current ovpn zip file is more than two hours old.
+  echo "$(adddate) INFO: Downloading new OVPN files."
 
-#First remove files if exists
-rm -rf ${OVPN_CONFIG_DIR}/ovpn*
+  #Remove current zip
+  rm -rf ${OVPN_CONFIG_DIR}/ovpn.zip
 
-#Curl download ovpn files from NordVPN
-curl -s -o ${OVPN_CONFIG_DIR}/ovpn.zip ${OVPN_FILES}
+  #Create directory if no volume is done
+  mkdir -p /app/ovpn/config
```

Comment:
The env var `OVPN_CONFIG_DIR` is `/app/ovpn/config` (which is mounted), so this should either be `mkdir -p ${OVPN_CONFIG_DIR}` or removed altogether because the path is mounted and therefore should exist. I know it's code which is present in the original source. :-)

Reply:
To create the folder is valid when there is no mount. So when you run the image with no binding it will fail. I agree on you suggention of using the env var 👍